SwaglyricsBot/general_commands.py

```diff
@@ -65,46 +75,41 @@ async def get_lyrics_command(self, ctx, song=None, artists=None):
                 tmp = artists
                 artists = list()
                 artists.append(tmp)
-
             artists_string = self.artists_to_string(artists)
             debug_string = "Getting lyrics for {} by {}".format(song, artists_string)
-
             await log.add_sub_log(debug_string)
             await ctx.send(debug_string)
 
-            lyrics = self.get_lyrics(song, artists[0])
-            await log.add_sub_log("Lyrics fetched successfully, splitting it into fields...")
-            splitted_lyrics = self.chop_string_into_chunks(lyrics, 1024)
-            await log.add_sub_log("Splitted successfully. Packing into messagess...")
-            
-            await self.send_chunks(ctx, splitted_lyrics, song, artists_string)
-            await log.add_sub_log(f"Lyrics sent successfully.", ConsoleColors.OKGREEN)
-            log.change_log_success_status(True)
+            send_lyrics()
         except LyricsError as ex:
             await log.add_sub_log(f"Error raised: {ex}", ConsoleColors.FAIL)
             log.change_log_success_status(None)
             await ctx.send(ex)
+        except AttributeError as ex:
+            # basically a bug where sometimes getting lyrics glitches and then it works in the retry
+            await log.add_sub_log(f"Error: {ex}", ConsoleColors.FAIL, True)
+            print(traceback.print_exception(type(ex), ex, ex.__traceback__))
+            log.change_log_success_status(False)
+
+            send_lyrics()

```

Comment:
This is all right, but we need to restrict the number of tries, for now, it usually fails only once, but we don't know if it won't fail more than once for the same request, to prevent overload and bottleneck add a number of tries and raise error after exceeding them.

Reply:
Add a retry amount for Attribute Error and add the number of retries to log, so we can track it. Let's say that for now, It'll try 2 times before raising an exception.